### ISSUE TITLE
Windows 10 build failing 64bit and Python error thrown missing libraries 

### ISSUE THREAD

**Issue by TakesTheBiscuit** [SFTtech/openage#1090]
Hi all

Windows 10 here. Sorry.

```
OS: Windows 10 Pro 64bit
Python: 3.7
Build: 64 bit
Python path: (i'm using specifying on command line to make python finder error go away):

DPYTHON="C:\Users\Admin\AppData\Local\Programs\Python\Python37\python.exe" )

Main issue: 
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
PYTHON_LIBRARIES
```

Digging into this a bit; with my brand new CMAKE and Python skillz;
I can see `${PYTHON_LIBRARIES}` currently equals `PYTHON_LIBRARIES-NOTFOUND`

I'll keep looking to see if i can find how python and cmake are getting this strange string, to me it looks like concatenated strings of `PYTHON_LIBRARIES` and `NOTFOUND`.

This part is fairly ugly on Windoze actually; if i find something before you see this i'll try my best to recommend the fix or i can always submit a PR.

Some eye candy;

![image](https://user-images.githubusercontent.com/15122732/53532360-820c9180-3aee-11e9-8308-d020172cef4a.png)

![image](https://user-images.githubusercontent.com/15122732/53532935-c39e3c00-3af0-11e9-9068-42c762d06809.png)

**Comment by TakesTheBiscuit** [comment]
### TLDR; Make sure to delete /bin/* when you make changes to python on windows as it caches the paths and versions etc


This is likely because the debug option(s) was not originally selected when python was installed using the python installer from python.org.

Installed 3.8 alpha with all the debug checkboxes on (symbols etc - infact everything was ticked except PATH);

So if i had followed the instructions and not just said to myself 'hmm i already have Python!' i would've been just fine. Apologies for raising the issue.

Future readers/builders and myself; Please kindly RTFM?

Anyone else struggling i had a few mini hurdles;
- had to add to path in windows and remove others - i had 2 other versions of python i've played with over the years
- totally installed an alpha of Python 3.8 can't see any issues
- had to restart powershell after changing path
- must run powershell as admin (elevated permissions to make folders/directories it's like unix 'sudo')
- got frustrated, uninstalled older versions of python completely
- got more annoyed when i found this didn't help ... deleted cached cmake build files inside /build
- level 12/10 frustration, found that `C:/Users/Admin/AppData/Local/Programs/Python/Python37` wasn't uninstalled even though i used the official uninstaller; no idea how openage is even finding this version of python at this point; deleted the whole directory manually and everything in it
- now make is complaining it cant find any version of python .. which is actually hilarious because i'm executing `python ./configure` USING python! 
- i gave up and hardcoded my path to Python in `FindPython.cmake` - apologies. Again Windows paths are horrid so had to escape my path like `c:\\x\\y\\z`
- spotted that the output was looking here:
```
FindPython_test.cpp
  cl /c /I"C:\Program Files\Python38\include"
```
Which isn't right... on windows the lib files are in
`C:\Program Files\Python38\libs`

- Copied contents of `\libs` to `\include` just to see if that did it
- nope.
- descending through the 10 levels of hell;
```
now have this in findpython.cmake:
set(PYTHON "C:\\Program Files\\Python38\\python.exe")
set(PYTHON_EXECUTABLE "C:\\Program Files\\Python38\\python.exe")
```

it's 2am. i started this at 23:00.
turns out tired brains don't work so well.
Undid all the crazy above, found `.bin` folder on windows. found a horrible cache file inside it - nuked it. Winning.

**Comment by TheJJ** [comment]
oh no! that sound horrible. I'm wondering though why it did not find python successfully in the beginning (independent of the cache folder). It would likely help other windows users if you figured out how our `FindPython.cmake` needs to be adapted to suceed.

Also, CMake has a `FindPython3` since version v3.12. We should consider using it instead, it pretty much does what our custom version did when CMake only shipped their crappy `FindPython` variant before 3.12.

**Comment by TakesTheBiscuit** [comment]
Hey @TheJJ - yeah pretty nasty build experience on Windows; i'm not being critical as i know Windows is kind of irrelevant right now.

I'm always pretty busy with work-work but i am thinking of ways i can help from a Windows angle.. potentially thinking how i can help build a virtual machine image file (docker perhaps) which comes with all the build tools ready to go so it's just a case of `git clone` then a call to make
